### ISSUE TITLE
Breaking change in Microsoft.IdentityModel - exception throw when calling IsNullOrEmpty() 

### ISSUE THREAD

**Issue by peterharding93** [alexhiggins732/IdentityServer8#40]
A breaking change was made in Microsoft.IdentityModel 8.0.0 was made which makes the extension method IsNullOrEmpty() private.  This method seems to have been made private at one of the 8.0.0 preview releases.

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2651

It seems that the nuget package IdentityServer8 version 8.0.4 tries to call this method.  This causes a run time exception.

I specifically hit this bug inside DefaultKeyMaterialService.GetSigningCredentialsAsync(), but there are a few other locations that seem to call this.

I made a workaround by explicit checking for null && !Any().
A quick search for all calls to IsNullOrEmpty should show all the locations where this code needs to be fixed.

**Comment by alexhiggins732** [comment]
It would be helpful if someone could open a pull request. I plan on working on some issues this weekend and if this isn't to involved I will try to get it in vNext

**Comment by quickstar** [comment]
> It would be helpful if someone could open a pull request. I plan on working on some issues this weekend and if this isn't to involved I will try to get it in vNext

This seems already somewhat merged in https://github.com/alexhiggins732/IdentityServer8/pull/41, right?

**Comment by joe-knd** [comment]
> > It would be helpful if someone could open a pull request. I plan on working on some issues this weekend and if this isn't to involved I will try to get it in vNext
> 
> This seems already somewhat merged in [#41](https://github.com/alexhiggins732/IdentityServer8/pull/41), right?

Is this already released? I see only 8.0.4 from last week but that still have the issue with the IsNullOrEmpty call

**Comment by quickstar** [comment]
> Is this already released? I see only 8.0.4 from last week but that still have the issue with the IsNullOrEmpty call

Unfortunately not yet! I'm waiting eagerly for the `8.0.5` which should fix those issues!

**Comment by quickstar** [comment]
> It would be helpful if someone could open a pull request. I plan on working on some issues this weekend and if this isn't to involved I will try to get it in vNext

@alexhiggins732 this is no longer required because it seems already merged, correct?

**Comment by joe-knd** [comment]
> > Is this already released? I see only 8.0.4 from last week but that still have the issue with the IsNullOrEmpty call
> 
> Unfortunately not yet! I'm waiting eagerly for the `8.0.5` which should fix those issues!

but the fix is already in the source code? could I use for now directly the projects instead of the package?

**Comment by quickstar** [comment]
> but the fix is already in the source code? could I use for now directly the projects instead of the package?

It looks like it.

**Comment by garethdhopkins** [comment]
We're just sorting out the release workflows in #32 now and we should be able to get v8.0.5 released and pushed to nuget soon.

**Comment by joe-knd** [comment]
Hi, is there any progress in the new release 8.0.5?

**Comment by JohanHeyvaert** [comment]
We're also experiencing this problem. 
It occurs when we add package Microsoft.Identity.Web in the project. We need this to connect to Azure (e.g. to use Microsoft Graph, in order to look up user info).

Hopefully this can quickly be solved? Many thanks in advance!

UPDATE:
Just to confirm that the problem is solved when using the source code of IdentityServer8.

**Comment by joe-knd** [comment]
Yeah, the source code has the fix already, but no official package yet 8.0.5

**Comment by sebastienlabine** [comment]
@alexhiggins732 Possible to just release this (without anything else)? Our current solution is to downgrade everything. This has been going on for a long time. Thanks

**Comment by paul-segerink** [comment]
Bump. I'd love an 8.0.5 as well. Kudo's for this entire venture.

**Comment by KellyTitan** [comment]
Just ran into this issue. Need 8.0.5 too. :(

**Comment by SocialMatrix** [comment]
We're also experiencing this problem. We need 8.0.5 as well. Is there any timeline for 8.0.5 release? TIA

**Comment by SocialMatrix** [comment]
@alexhiggins732 Is there a way we can assist in getting the 8.0.5 out sooner?

**Comment by joe-knd** [comment]
any news about a8.0.5 release? the fix is the code but there is no new nuget package for this yet

**Comment by techadvisorbe** [comment]
I couldn’t wait, so I created a NuGet package based on the official source (forked):
https://www.nuget.org/packages/TechAdvisor.IdentityServer8/8.0.5

I hope the maintainer will either upgrade the official package to version .NET 10 or publish version 8.0.5 on the existing NuGet.
